pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -634,8 +643,19 @@ def initialize(self):
         # setup spoke state based on data from the Subscription DBus module
         self._update_spoke_state()
 
+        # start the rest of spoke initialization which might take some time
+        # (mainly due to waiting for various initialization threads to finish)
+        # in a separate thread
+        threadMgr.add(AnacondaThread(name=THREAD_SUBSCRIPTION_SPOKE_INIT,
+                                     target=self._initialize))
+
+    def _initialize(self):
         # wait for subscription thread to finish (if any)
         threadMgr.wait(THREAD_SUBSCRIPTION)
+        # also wait for the payload thread, which migh still be processing
+        # a CDROM source, to avoid the Subscription being mandatory by mistake
+        # due to CDN still being default at the time of evaulation
+        threadMgr.wait(THREAD_PAYLOAD)
 
         # update overall state
         self._update_registration_state()
```

Comment:
Why do this twice?

Reply:
I'm not sure I understand, these are different methods:

```
        self._update_registration_state()
        self._update_subscription_state()
```

Or do you mean something else ?

Reply:
Oh, wow, they are :O never mind then...